File: loyalty-service/src/main/java/com/javacaptain/reward/infrastructure/MonitoringConfiguration.java
package com.javacaptain.reward.infrastructure;

import io.micrometer.core.instrument.MeterRegistry;
import org.springframework.boot.actuate.autoconfigure.metrics.MeterRegistryCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
class MonitoringConfiguration {

    @Bean
    public MeterRegistryCustomizer<MeterRegistry> commonTags(){
        return registry -> registry.config().commonTags("application", "loyalty-service");
    }
}

File: account-service/src/main/kotlin/com/javacaptain/eventsourcingaccount/infrastructure/MonitoringConfiguration.kt
package com.javacaptain.eventsourcingaccount.infrastructure

import io.micrometer.core.instrument.MeterRegistry
import org.springframework.boot.actuate.autoconfigure.metrics.MeterRegistryCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class MonitoringConfiguration {
    @Bean
    fun commonTags(): MeterRegistryCustomizer<MeterRegistry>? {
        return MeterRegistryCustomizer { registry: MeterRegistry ->
            registry.config().commonTags("app", "event-sourcing-account")
        }
    }
}
